include/os/freebsd/spl/sys/debug.h

```diff
@@ -54,9 +54,16 @@
 /*
  * Common DEBUG functionality.
  */
-int spl_panic(const char *file, const char *func, int line,
-    const char *fmt, ...);
-void spl_dumpstack(void);
+extern void spl_panic(const char *file, const char *func, int line,
+    const char *fmt, ...) __attribute__((format(printf, 4, 5), __noreturn__));
+extern void spl_dumpstack(void);
+
+static inline int
+spl_assert(const char *buf, const char *file, const char *func, int line)
+{
+	spl_panic(file, func, line, "%s", buf);
+	return (0);
+}
```

Comment:
I'm fine with adding an `spl_assert()` wrapper, but it's not clear to me why this was needed.

Reply:
Build failures occurred with other variations of this. I ended up having to mirror the userland version to get it to work.

Reply:
For example, GCC will print a warning that -Werror turns into an error if you set `__attribute__((__noreturn__))` on a function that returns an `int`. At the same time, we have VERIFY and ASSERT in certain preprocessor macros that require that they return an int (and ignores the `(void)` in our macros).

Reply:
Could we instead change the VERIFY definition to be an `if` statement rather than a boolean expression?

Reply:
Not without breaking several uses of them in the codebase.